src/Services/DateService.php

```diff
@@ -0,0 +1,19 @@
+<?php
+namespace ProjectManager\Services;
+
+class DateService
+{
+    public static function isOverdue(string $date): bool
+    {
+        $deadline = new \DateTimeImmutable($date);
+        $currentDate = new \DateTimeImmutable();
+        $dateDifference = $deadline->diff($currentDate);
+        return !$dateDifference->invert;
```

Comment:
this could be shortened with objecy chaining by accessing `->invert` immediately on the `->diff`